engines/macventure/macventure.cpp

```diff
@@ -958,9 +958,18 @@ Common::String MacVentureEngine::getCommandsPausedString() const {
 
 Common::Path MacVentureEngine::getFilePath(FilePathID id) const {
 	if (id <= 3) { // We don't want a file in the subdirectory
+
+		// Convert from MacRoman encoding to unicode for punycode encoding
+		Common::String result = Common::convertToU32String((_filenames->getString(id)).c_str(), Common::kMacRoman);
```

Comment:
No, instead of converting both names to U32, you need to convert to Punycode the original filenames.

E.g. something like this:
```C++
Common::punycode_encodefilename(fileName)
```